package/MDAnalysis/analysis/rms.py

```diff
@@ -535,29 +541,52 @@ def __init__(self, atomgroup, reference=None, select='all',
                         igroup, sel['reference'], sel['mobile'],
                         len(atoms['reference']), len(atoms['mobile'])))
 
-        # Explicitly check for "mass" because this option CAN
-        # be used with groupselection. (get_weights() returns the mass array
-        # for "mass")
+        # check weights type
         if not iterable(self.weights) and self.weights == "mass":
             pass
         else:
-            self.weights = get_weights(self.mobile_atoms, self.weights)
+            get_weights(self.mobile_atoms, self.weights)
+
+        if self.weights_groupselections:
+            if len(self.weights_groupselections) != len(self.groupselections):
+                raise ValueError("Length of weights_groupselections is not equal to "
+                                 "length of groupselections ")
+            for weights, atoms, selection in zip(self.weights_groupselections,
+                                                 self._groupselections_atoms,
+                                                 self.groupselections):
+                try:
+                    if not iterable(weights) and weights == "mass":
+                        pass
+                    else:
+                        get_weights(atoms['mobile'], weights)
+                except Exception as e:
+                    raise type(e)(str(e) + ' happens in selection %s' % selection['mobile'])
 
-        # cannot use arbitrary weight array (for superposition) with
-        # groupselections because arrays will not match
-        if (len(self.groupselections) > 0 and (
-                iterable(self.weights) or self.weights not in ("mass", None))):
-            raise ValueError("groupselections can only be combined with "
-                             "weights=None or weights='mass', not a weight "
-                             "array.")
 
     def _prepare(self):
         self._n_atoms = self.mobile_atoms.n_atoms
-
-        if not iterable(self.weights) and self.weights == 'mass':
-            self.weights = self.ref_atoms.masses
-        if self.weights is not None:
-            self.weights = np.asarray(self.weights, dtype=np.float64) / np.mean(self.weights)
+        if not self.weights_groupselections:
+            if not iterable(self.weights):         # apply 'mass' or 'None' to groupselections
+                self.weights_groupselections = [self.weights] * len(self.groupselections)
+            else:
+                self.weights_groupselections = [None] * len(self.groupselections)
+
+        for igroup, (weights, atoms) in enumerate(zip(self.weights_groupselections,
+                                                      self._groupselections_atoms), 0):
+            if weights == 'mass':
```

Comment:
if weights is a numerical numpy array like masses this will raise 
```
`FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison`. 
```

We should probably catch this warning to avoid confusion.